### PR TITLE
Some without std::shared_ptr

### PR DESCRIPTION
**Pull request by lisitsyn**
Ok guys lets discuss if this is better

### CODE REVIEW

**Comment by lambday** [comment]
@lisitsyn why not using c++11 version?

File: src/shogun/base/some.h [hunk 1]
```diff
@@ -41,62 +39,164 @@ namespace shogun
...
+		__CREATE_SOME(__PARAMETERS_2(um, dois));
+	}
+	template <typename T, typename Um, typename Dois, typename Tres>
+	inline Some<T> some(const Um& um, const Dois& dois, const Tres& tres)
+	{
+		__CREATE_SOME(__PARAMETERS_3(um, dois, tres));
+	}
+	template <typename T, typename Um, typename Dois, typename Tres, typename Quatro>
+	inline Some<T> some(const Um& um, const Dois& dois, const Tres& tres, const Quatro& quatro)
+	{
```
Comment (karlnapf) [inline src/shogun/base/some.h]: If this is the cost of not having c++, I definitely vote for c++11. This is too messy to read (at least for me)

Comment (lisitsyn) [inline src/shogun/base/some.h]: Well it is just a cascade of all possible sizes

The main mess comes from no shared_ptr being used

**Comment by lisitsyn** [comment]
@lambday to not force all the shogun be C++11 :) Just an idea

File: src/shogun/base/some.h (hunk 1 above)
Comment (lambday) [inline src/shogun/base/some.h]: @lisitsyn only up to 5 constructor params - 5 should suffice and rest should be initialized by setters but still variadic is good - why restrict ? I will also vote for c++11 one :P

**Comment by iglesias** [comment]
I think the methods in Some (constructor, assign, ref, unref, etc) might cause us terrible headaches in the future :-)

How bad would it be to make C++11 a requirement from a certain version of Shogun? I really have no idea whether it is just nuts doing that. What do other open source projects usually do? Are there important compilers for which we would directly lose support?

File: src/shogun/base/some.h (hunk 1 above)
Comment (karlnapf) [inline src/shogun/base/some.h]: As we kind of agreed on c++11, lets DO it :)

**Pull request closed without merging**